### ISSUE TITLE
Bug: `merge` does not merge ObjectLikeSequence

### ISSUE THREAD

**Issue by Redsandro** [dtao/lazy.js#169]
According to [the docs](http://danieltao.com/lazy.js/docs/#ObjectLikeSequence-merge):

> `ObjectLikeSequence.merge = function(others, mergeFn)`
>
> `others`	...Object|__ObjectLikeSequence__
The other object(s) __or sequence(s)__ whose values will be merged into this one.

However,

```
var a = Lazy({kaas: {lekker: true}});
var b = Lazy({kaas: {gouds: true}});
a.merge(b).value();
```

Expected:
```
{
    kaas: {
        lekker: true,
        gouds: true
    }
}
```

Actual:

```
{
    kaas: {
        lekker: true
    },
    source: {
        kaas: {
            gouds: true
        }
    }
}
// This object also has all of Lazy.js ObjectLikeSequence functions attached.
```

**Comment by Redsandro** [comment]
I think this needs fixing in [jazy.js:3958](https://github.com/dtao/lazy.js/blob/master/lazy.js#L3958) but this code is way to magical for me to understand how to enumerate the lazily contained properties in stead of the javascript wrapper properties.

```javascript
    for (prop in a) {
      merged[prop] = mergeObjects(a[prop], b[prop]);
    }
    for (prop in b) {
      if (!merged[prop]) {
        merged[prop] = b[prop];
      }
    }
```

**Comment by Redsandro** [comment]
@dtao I think this needs fixing in [jazy.js:3958](https://github.com/dtao/lazy.js/blob/master/lazy.js#L3958) but this code is way to magical for me to understand how to enumerate the lazily contained properties in stead of the javascript wrapper properties.

```javascript
    for (prop in a) {
      merged[prop] = mergeObjects(a[prop], b[prop]);
    }
    for (prop in b) {
      if (!merged[prop]) {
        merged[prop] = b[prop];
      }
    }
```

I'm guessing something with a `switch` and `b.get(prop)` and `b.each()`.

**Comment by dtao** [comment]
I agree that this is a bug in that it doesn't work as advertised. Unfortunately, looking at the merge code right now... yeah, seems like it might be pretty annoying to fix. I'll try to get to it, but probably not tonight.
